src/snowflake/snowpark/dataframe.py

```diff
@@ -1075,13 +1075,24 @@ def to_pandas(
                 **kwargs,
             )
 
-        # if the returned result is not a pandas dataframe, raise Exception
-        # this might happen when calling this method with non-select commands
-        # e.g., session.sql("create ...").to_pandas()
         if block:
             if not isinstance(result, pandas.DataFrame):
+                query = self._plan.queries[-1].sql.strip().lower()
+                is_select_statement = is_sql_select_statement(query)
+                if is_select_statement:
```

Comment:
can we have another check here checking result format being JSON? the info should be available in the connection session parameters dict